packages/legacy/core/App/components/misc/InactivityWrapper.tsx

```diff
@@ -0,0 +1,121 @@
+import { useAgent } from '@credo-ts/react-hooks'
+import React, { PropsWithChildren, useEffect, useRef } from 'react'
+import { AppState, PanResponder, View } from 'react-native'
+import { useAuth } from '../../contexts/auth'
+import { useStore } from '../../contexts/store'
+import { DispatchAction } from '../../contexts/reducers/store'
+import { TOKENS, useServices } from '../../container-api'
+
+export enum LockOutTime {
+  OneMinute = 1,
+  ThreeMinutes = 3,
+  FiveMinutes = 1,
+  Never = 0,
+}
+
+interface InactivityWrapperProps {
+  timeoutLength?: LockOutTime // number of minutes before timeoutAction is triggered, a value of 0 will never trigger the timeoutAction and an undefined value will default to 5 minutes
+}
+
+const InactivityWrapper: React.FC<PropsWithChildren<InactivityWrapperProps>> = ({ children, timeoutLength }) => {
+  const [logger] = useServices([TOKENS.UTIL_LOGGER])
+  useStore()
+  const [_, dispatch] = useStore()
+  const { agent } = useAgent()
+  const { removeSavedWalletSecret } = useAuth()
+  const timeout_minutes = timeoutLength !== undefined ? timeoutLength : LockOutTime.OneMinute
+  const inactivityTimer = useRef<NodeJS.Timeout | null>(null)
+  const panResponder = React.useRef(
+    PanResponder.create({
+      onStartShouldSetPanResponderCapture: () => {
+        // some user interaction, reset timeout
+        resetInactivityTimeout(timeout_minutes)
+
+        // returns false so the PanResponder doesn't consume the touch event
+        return false
+      },
+    })
+  ).current
+
+  const resetInactivityTimeout = (minutes: number) => {
+    // remove existing timeout
+    clearTimer()
+
+    // do not start timer if timeout is set to 0
+    if (minutes > 0) {
+      // create new timeout
+      inactivityTimer.current = setTimeout(async () => {
+        lockUserOut()
+      }, minutesToMilliseconds(minutes))
+    }
+  }
+
+  const lockUserOut = async () => {
+    try {
+      removeSavedWalletSecret()
+      await agent?.wallet.close()
+    } catch (error) {
+      logger.error(`Error closing agent wallet, ${error}`)
+    }
+
+    dispatch({
+      type: DispatchAction.DID_AUTHENTICATE,
+      payload: [{ didAuthenticate: false }],
+    })
+
+    dispatch({
+      type: DispatchAction.LOCKOUT_UPDATED,
+      payload: [{ displayNotification: true }],
+    })
+  }
+
+  const clearTimer = () => {
+    if (inactivityTimer.current) {
+      clearTimeout(inactivityTimer.current)
+    }
+  }
+
+  const minutesToMilliseconds = (minutes: number) => {
```

Comment:
I might just simplify this to `const blahAsMiliseconds = minutes * 60000` at the top of the file.

Reply:
good call